dali/operators/reader/loader/video/frames_decoder_test.cc

```diff
@@ -0,0 +1,143 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <exception>
+
+#include "dali/test/dali_test_config.h"
+#include "dali/core/error_handling.h"
+#include "dali/operators/reader/loader/video/video_test_base.h"
+#include "dali/operators/reader/loader/video/frames_decoder.h"
+
+
+namespace dali {
+class FramesDecoderTest : public VideoTestBase {
+};
+
+
+TEST_F(FramesDecoderTest, ConstantFrameRate) {
+    std::string path = testing::dali_extra_path() + "/db/video/cfr/test_1.mp4";
+
+    // Create file, build index
+    FramesDecoder file(path);
+
+    ASSERT_EQ(file.Height(), 720);
+    ASSERT_EQ(file.Width(), 1280);
+    ASSERT_EQ(file.Channels(), 3);
+    ASSERT_EQ(file.NumFrames(), 50);
+
+    std::vector<uint8_t> frame(file.FrameSize());
+
+    // Read first frame
+    file.ReadNextFrame(frame.data());
+    this->ComapreFrames(frame.data(), this->GetCfrFrame(0, 0), file.FrameSize());
+
+    // Seek to frame
+    file.SeekFrame(25);
+    file.ReadNextFrame(frame.data());
+    this->ComapreFrames(frame.data(), this->GetCfrFrame(0, 25), file.FrameSize());
+
+    // Seek back to frame
+    file.SeekFrame(12);
+    file.ReadNextFrame(frame.data());
+    this->ComapreFrames(frame.data(), this->GetCfrFrame(0, 12), file.FrameSize());
+
+    // Seek to last frame (flush frame)
+    file.SeekFrame(49);
+    file.ReadNextFrame(frame.data());
+    this->ComapreFrames(frame.data(), this->GetCfrFrame(0, 49), file.FrameSize());
+
+    // Wrap around to first frame
+    ASSERT_FALSE(file.ReadNextFrame(frame.data()));
+    file.Reset();
+    file.ReadNextFrame(frame.data());
+    this->ComapreFrames(frame.data(), this->GetCfrFrame(0, 0), file.FrameSize());
+}
+
+TEST_F(FramesDecoderTest, VariableFrameRate) {
+    std::string path = testing::dali_extra_path() + "/db/video/vfr/test_2.mp4";
+
+    // Create file, build index
+    FramesDecoder file(path);
+
+    ASSERT_EQ(file.Height(), 600);
+    ASSERT_EQ(file.Width(), 800);
+    ASSERT_EQ(file.Channels(), 3);
+    ASSERT_EQ(file.NumFrames(), 60);
+
+    std::vector<uint8_t> frame(file.FrameSize());
+
+    // Read first frame
+    file.ReadNextFrame(frame.data());
+    this->ComapreFrames(frame.data(), this->GetVfrFrame(1, 0), file.FrameSize());
+
+    // Seek to frame
+    file.SeekFrame(25);
+    file.ReadNextFrame(frame.data());
+    this->ComapreFrames(frame.data(), this->GetVfrFrame(1, 25), file.FrameSize());
+
+    // Seek back to frame
+    file.SeekFrame(12);
+    file.ReadNextFrame(frame.data());
+    this->ComapreFrames(frame.data(), this->GetVfrFrame(1, 12), file.FrameSize());
+
+    // Seek to last frame (flush frame)
+    file.SeekFrame(59);
+    file.ReadNextFrame(frame.data());
+    this->ComapreFrames(frame.data(), this->GetVfrFrame(1, 59), file.FrameSize());
+
+    // Wrap around to first frame
+    ASSERT_FALSE(file.ReadNextFrame(frame.data()));
```

Comment:
I thought it's been removed - stale comment?

Reply:
Same as above